cross_val.py

```diff
@@ -107,66 +105,59 @@ def _train(self, criterion, optimizer, fold_idx):
+                    writer.add_scalar('train_loss', loss, global_step)
+                    writer.add_scalar('val_loss', val_loss, global_step)
+                    print('train_loss: {:.4f}, val_loss: {:.4f}'.format(loss.item(), val_loss))
 
         writer.close()
-        return max_test_acc, max_f1
+        return np.mean(val_loss_list)
 
-    def _evaluate_acc_f1(self, data_loader):
-        n_test_correct, n_test_total = 0, 0
-        t_targets_all, t_outputs_all = None, None
+    def _evaluate_loss(self, criterion, data_loader):
+        v_loss_total = 0
+        n_val_total = 0
         with torch.no_grad():
-            for t_batch, t_sample_batched in enumerate(data_loader):
-                t_inputs = [t_sample_batched[col].to(opt.device) for col in self.opt.inputs_cols]
-                t_targets = t_sample_batched['polarity'].to(opt.device)
-                t_outputs = self.model(t_inputs)
+            for v_batch, v_sample_batched in enumerate(data_loader):
+                v_inputs = [v_sample_batched[col].to(opt.device) for col in self.opt.inputs_cols]
+                v_targets = v_sample_batched['polarity'].to(opt.device)
+                v_outputs = self.model(v_inputs)
 
-                n_test_correct += (torch.argmax(t_outputs, -1) == t_targets).sum().item()
-                n_test_total += len(t_outputs)
+                v_loss = criterion(v_outputs, v_targets)
+                v_loss_total += v_loss.item()
+                n_val_total += len(v_outputs)
 
-                if t_targets_all is None:
-                    t_targets_all = t_targets
-                    t_outputs_all = t_outputs
-                else:
-                    t_targets_all = torch.cat((t_targets_all, t_targets), dim=0)
-                    t_outputs_all = torch.cat((t_outputs_all, t_outputs), dim=0)
+        val_loss = v_loss_total / n_val_total
+        return val_loss
 
-        test_acc = n_test_correct / n_test_total
-        f1 = metrics.f1_score(t_targets_all.cpu(), torch.argmax(t_outputs_all, -1).cpu(), labels=[0, 1, 2], average='macro')
-        return test_acc, f1
-
-    def run(self):
+    def run(self, repeats=5):
         # Loss and Optimizer
         criterion = nn.CrossEntropyLoss()
         _params = filter(lambda p: p.requires_grad, self.model.parameters())
         optimizer = self.opt.optimizer(_params, lr=self.opt.learning_rate, weight_decay=self.opt.l2reg)
-        
-        for i in range(self.opt.fold):
-            print('fold: '+str(i+1))
-            self._reset_params()
-            max_test_acc, max_f1 = self._train(criterion, optimizer, i)
-            print('max_test_acc: {0}     max_f1: {1}'.format(max_test_acc, max_f1))
 
+        best_val_loss = float('inf')
+        for r in range(repeats):
+            print('repeat: '+str(r+1))
+            fold_avg_val_losses = []
+            for i in range(self.opt.fold):
+                print('fold: '+str(i+1))
+                self._reset_params()
```

Comment:
For BERT-based models, the usage of `reset_params` in for loops may be problematic, since we don't reset BERT params in  `reset_params` and fine-tuning of BERT parameters will be retained.

Reply:
But I found that in `reset_params` there exists related judgement that BertModel won't reset their parameters.

Reply:
That's the problem. The fine-tuning of BERT in the previous fold is reserved for the next fold, resulting in cumulative results.

Reply:
So should I reset the parameters or not ? As I'm not so familiar with BERT...

Reply:
I‘ll do this in new PR.